assigner/config.py

```diff
@@ -136,10 +136,8 @@ def __getattr__(self, key):
         # Keys contained dashes can be called using an underscore
         key = key.replace("_", "-")
 
-        try:
-            return self.data[key]
-        except KeyError as e:
-            if key == "roster":
-                self.data[key] = []
-                return self.data[key]
-            raise e
+        # Fill in a blank roster if needed
+        if key == "roster" and (key not in self.data or self.data[key] is None):
```

Comment:
Is self.data a property? I can't remember. My intuition here is to tell you to use `self.data.get(key, None)` or `self.data.get(key)` which defaults to `None` (or it did in Python 2).

Reply:
Calling `self.data.get(key, [])` doesn't set `self.data[key] = []`, so any modifications to the empty roster list returned wouldn't actually show up in the config.

Example:
```
>>> data = {}
>>> l = data.get("bob", [])
>>> l
[]
>>> data
{}
>>> l.append(5)
>>> l
[5]
>>> data
{}
```

Reply:
Ohh, I see what you mean. Yeah, I'll change the condition--yours is a lot nicer. ❤️